### PR TITLE
Add a Slack user cache

### PR DESCRIPTION
**Pull request by ChrisAnn**
Part of the work for pre-loading all external users to the database.

Tested locally to ensure all users are added to the database and returned from the users api endpoint.

### CODE REVIEW

File: response/core/models/user_external.py [hunk 1]
```diff
@@ -11,6 +10,7 @@ class Meta:
     app_id = models.CharField(max_length=50, blank=False, null=False)
     external_id = models.CharField(max_length=50, blank=False, null=False)
     display_name = models.CharField(max_length=50, blank=False, null=False)
+    full_name = models.CharField(max_length=50, blank=True, null=True)
```
Comment (ChrisAnn) [inline response/core/models/user_external.py]: This was added to consolidate slack profile data returning `fullname` and the e`ExternalUser` models not including this.  Having read the Slack docs, `fullname != display_name` so we can't substitute one for another.

File: response/slack/cache.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                        'full_name': user["profile"]["real_name"] or user['name'],
+                    },
+                )
+        cursor = response['response_metadata'].get('next_cursor')
+
+        if not cursor:
+            break
+
+
+def get_user_profile(external_id):
```
Comment (ChrisAnn) [inline response/slack/cache.py]: In another PR, we should replace all usages of settings.SLACK_CLIENT.get_user_profile with this to take advantage of the cache.

File: response/core/models/user_external.py (hunk 1 above)
Comment (milesbxf) [reply response/core/models/user_external.py]: Interesting. Any idea how they differ? Should we consider also returning `full_name` in the API response?

File: response/slack/cache.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+from response.core.models import ExternalUser, GetOrCreateSlackExternalUser
+from response.slack.client import SlackError
+
+logger = logging.getLogger(__name__)
+
+
+def update_user_cache():
+    cursor = None
+    while True:
```
Comment (milesbxf) [inline response/slack/cache.py]: You could do `while cursor != "":` here and remove the check below (I've just remembered that https://github.com/monzo/response/pull/85 does this)

File: tests/slack/slack_payloads.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            "is_app_user": "False",
+            "has_2fa": "False"
+        },
+        {
+            "id": "U12345678",
+            "team_id": "T0G9PQBBK",
+            "name": "chrisevans",
+            "deleted": "False",
+            "color": "9f69e7",
+            "real_name": "Glinda Southgood",
```
Comment (milesbxf) [inline tests/slack/slack_payloads.py]: 😮 @evnsio you never told us about your real name!

File: response/core/models/user_external.py (hunk 1 above)
Comment (ChrisAnn) [reply response/core/models/user_external.py]: The difference seems to be this: https://get.slack.help/hc/en-gb/articles/115004692303-Change-how-names-are-displayed
We probably want `full_name` on the UI, so yeah, I'll add it to the API response.

File: tests/slack/slack_payloads.py (hunk 4 above)
Comment (ChrisAnn) [reply tests/slack/slack_payloads.py]: Dammit 🤦‍♀ .  I thought I'd removed all of mention of Chris in the test data.  Sorry :(

**Pull request merged**